pkg/api/v1beta4/dynakube/oneagent/props.go

```diff
@@ -104,7 +105,7 @@ func (oa *OneAgent) GetConnectionInfoConfigMapName() string {
 }
 
 func (oa *OneAgent) IsReadOnlyOneAgentsMode() bool {
```

Comment:
It is not really readonly :D

Because it can write to the CSI volume (which is a "hidden" hostPath)
or to what the `StorageHostPath` is pointing to.

I would either call it `NeedsDedicatedStorage` or something like that:
- `NeedsStorageVolume`
- `NeedsHostStorage`
- ...

(if you don't like any of them, then we can keep it as is, don't need to change it for the sake of changing it 😉 )

Reply:
maybe `NeedsVolumeStorage`, `IsVolumeStorageMode` or `UsesVolumeStorage` 👀 ?

Reply:
my 2 cents: I personally would tend to `NeedsVolumeStorage` :)

Reply:
I would prefer to clearly say whether file system is RO or RW
```
IsOneAgentModeSupportingReadOnlyFS() {
	return oa.IsCloudNativeFullstackMode() || oa.IsHostMonitoringMode()
}
```
contrary to ClassicFullStack which requires RW filesystem.

See the function:
```
// isRootFsReadonly checks if the given version of the OneAgent supports the `ReadOnlyRootFilesystem` securityContext setting.
// if the version is not set, ie.: unknown, we  consider the OneAgent to support `ReadOnlyRootFilesystem`.
func (b *builder) isRootFsReadonly() bool {
	if b.dk != nil &&
		b.dk.OneAgent().IsReadOnlyOneAgentsMode() &&
		b.dk.OneAgent().GetVersion() != "" &&
		b.dk.OneAgent().GetVersion() != string(status.CustomImageVersionSource) {
```

Reply:
[commit](https://github.com/Dynatrace/dynatrace-operator/pull/4632/commits/c3dbe95e06a37897e68b2f212d55878db995c772)